### PR TITLE
Add android AOT/PInvoke functional test

### PR DESCRIPTION
**Pull request by MaximLipnin**
1. `./build.sh mono+libs -os Android -arch x64 -c Release `
2.  `./dotnet.sh build /t:Test src/tests/FunctionalTests/Android/Device_Emulator/AOT /p:TargetOS=Android /p:TargetArchitecture=x64 /p:Configuration=Release`
3.  `./dotnet.sh build /t:Test src/tests/FunctionalTests/Android/Device_Emulator/PInvoke /p:TargetOS=Android /p:TargetArchitecture=x64 /p:Configuration=Release`

### CODE REVIEW

**Comment by steveisok** [comment]
@MaximLipnin You'll need to rebase against master in order for the aot compiler to warn as opposed to error.

File: eng/testing/tests.mobile.targets [hunk 1]
```diff
@@ -53,6 +62,17 @@
...
     <RemoveDir Directories="$(BundleDir)" />
+
+    <MonoAOTCompiler Condition="'$(RunAOTCompilation)' == 'true'"
+        CompilerBinaryPath="$(MicrosoftNetCoreAppRuntimePackNativeDir)cross\$(PackageRID)\mono-aot-cross"
+        Mode="Full"
+        OutputType="AsmOnly"
+        Assemblies="@(AotInputAssemblies)"
+        UseLLVM="$(MonoEnableLLVM)"
+        LLVMPath="$(MicrosoftNetCoreAppRuntimePackNativeDir)cross\$(PackageRID)">
+        <Output TaskParameter="CompiledAssemblies" ItemName="BundleAssemblies" />
```
Comment (mdh1418) [inline eng/testing/tests.mobile.targets:73]: Are we using the `BundleAssemblies` item yet? Would it replace `SourceDir` which would require some tweaking to `AndroidAppBuilder`/`ApkBuilder`?

Comment (MaximLipnin) [reply eng/testing/tests.mobile.targets:73]: Yeah, thanks, I noticed it as well. But firstly it should start to build in AOT w/o errors :)

Comment (MaximLipnin) [reply eng/testing/tests.mobile.targets:73]: Mitchell, if you have any idea how to do it right, please let me know.

Comment (mdh1418) [reply eng/testing/tests.mobile.targets:73]: I'm not entirely sure what all needs to be changed. I'm thinking that the `AndroidAppBuilderTask` takes in all the assemblies in `$(PublishDir)` as `SourceDir` which 
1) Checks that `$(MainLibraryFileName)` exists. 
2) zip all the files to `apk/assets/assets.zip` https://github.com/dotnet/runtime/blob/5761dd49339911241209c1ffd61105a94edb6df1/src/tasks/AndroidAppBuilder/ApkBuilder.cs#L106-L107
3) makes sure  libmonosgen-2.0.a exists
4) adds the `.so` files to the dynamic library

It seems like using the AOT'd assemblies `BundleAssemblies` would need to cover those steps instead.
@EgorBo would you happen to know if this is the right approach?

Comment (MaximLipnin) [reply eng/testing/tests.mobile.targets:81]: Should be addressed now

**Review by steveisok — APPROVED** [review]
@akoeplinger You mentioned something needed to be done to the AOT task beyond this PR.  If that still applies, can you explain?

**Comment by steveisok** [comment]
@akoeplinger Please review and if it looks good, merge.

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**